piker/data/_web_bs.py

```diff
@@ -172,7 +172,9 @@ class JSONRPCResult(Struct):
 async def open_jsonrpc_session(
     url: str,
     start_id: int = 0,
-    dtype: type = JSONRPCResult
+    response_type: type = JSONRPCResult,
```

Comment:
As per above comment could we maybe make this a union of this result and the new `JSONRPCChannel`?

So it would be:
```python
response_type: JSONRPCChannel | JSONRPCResult
```

further, if you wanted to get fancier / more modern with this, instead of doing the try / except in the msg loop below you could use a new `match msg:` syntax and then match against the msg type?

Check docs for this here:
https://peps.python.org/pep-0636/#adding-a-ui-matching-objects

obviously in the `JSONRPCChannel` case it would just be ignored / logged or wtv.